tests/devices/test_default_mixed.py

```diff
@@ -39,105 +73,426 @@ def test_expected_state(self, nr_wires, index):
         rho = np.zeros((2 ** nr_wires, 2 ** nr_wires))
         rho[index, index] = 1
         rho = np.reshape(rho, [2] * (2 * nr_wires))
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert np.allclose(rho, dev._create_basis_state(index))
 
 
-@pytest.mark.parametrize("nr_wires", [1, 2, 3])
+@pytest.mark.parametrize("nr_wires", [2, 3])
 class TestState:
     """Tests for the method `state()`, which retrieves the state of the system"""
-    # Note: These tests need to be extended once we can output non-basis states
 
     def test_shape(self, nr_wires):
         """Tests that the state has the correct shape"""
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert (2 ** nr_wires, 2 ** nr_wires) == np.shape(dev.state)
 
     def test_init_state(self, nr_wires):
         """Tests that the state is |0...0><0...0| after initialization of the device"""
         rho = np.zeros((2 ** nr_wires, 2 ** nr_wires))
         rho[0, 0] = 1
-        dev = qml.device('default.mixed', wires=nr_wires)
+        dev = qml.device("default.mixed", wires=nr_wires)
 
         assert np.allclose(rho, dev.state)
 
-
-@pytest.mark.parametrize("nr_wires", [1, 2, 3])
+    @pytest.mark.parametrize("op", all_ops)
+    def test_state_after_op(self, nr_wires, op):
+        """Tests that state is correctly retrieved after applying operations on the first wires"""
+        dev = qml.device("default.mixed", wires=nr_wires)
+        if op == CNOT or op == CZ:
+            dev.apply([op(wires=[0, 1])])
+        elif op in channels:
+            # set channel parameter to 0.5
+            dev.apply([op(0.5, wires=0)])
+        else:
+            dev.apply([op(wires=0)])
+        current_state = np.reshape(dev._state, (2 ** nr_wires, 2 ** nr_wires))
+        assert np.allclose(dev.state, current_state)
```

Comment:
Any reason for not using `atol=tol, rtol=0` here?

Reply:
This is a good question. From the very beginning of SF and PL development, we always enforced by convention approximate equal assertions using the same value of absolute and relative tolerance.

The reasoning being that we could control, at a global level, the tolerance used across the entire test suite. I don't know if this is the right approach though, let me thinking about it.

P.S. I just discovered https://pypi.org/project/pytest-allclose/, which looks really cool.